cw/__init__.py

```diff
@@ -40,7 +40,14 @@
 
 
 def send(cfg, message):
-    return messaging.send(cfg, message)
+    conn = None
+    try:
+        conn = messaging.open_connection()
+        return messaging.send(cfg, message)
```

Comment:
based on your mods to messaging.send, dont you need to pass the newly created connection object (conn) as the 3rd arg?